ogarniacz_mp3.py

```diff
@@ -14,6 +14,33 @@ def get_music_dirs(dir_path):
     return music_dirs
 
 
+def reading_arguments_from_terminal():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--p2o")
+    parser.add_argument("--o2w")
+    parser.add_argument("--w2g")
+    args = parser.parse_args()
+    if args == "--p2o":
+        downloaded_to_processing()
+    if args == "--o2w":
+        processing_to_verification()
+    if args == "--w2g":
```

Comment:
Podejrzewam, że program się nie wywala przy uruchomieniu, ale czy jeśli w terminalu użyjesz `--w2g` to ten warunek zostanie spełniony? Sprawdź.

Reply:
Okej, więc program się nie wywala, ponieważ nie podaję żadnego argumentu. Gdy go podam, to się wywala i chce ode mnie jeszcze jeden argument, który będzie wartością obiektu opcjonalnego. Ale, jeśli dobrze zrozumiałam, my nie chcemy tam dawać żadnego argumentu.
Fragment kodu:
 ```
 if reading_arguments_from_terminal() == "--p2o":
        downloaded_to_processing()
```

oznacza, że by ```if``` zostało wykonane, to funkcja ```reading_arguments_from_terminal()``` musi zwrócić string ```"--p2o"```. Ta funkcja zwraca zmienną ```args```, która z kolei zawiera to, co zwrócił nam parser - czyli fragment ```parser.add_argument("--p2o", ...)``` trzeba napisać tak, by zwracał ```"--p2o"``` (oraz odpowiednie stringi do reszty argumentów).
Dobrze rozumiem?

Reply:
> Gdy go podam, to się wywala i chce ode mnie jeszcze jeden argument

Sprawdź [ten](https://realpython.com/command-line-interfaces-python-argparse/#setting-the-action-to-be-taken-for-an-argument) akapit. Zwróć uwagę na `store_true`.

Problem z `if args == "--o2w":` jest taki, że args zawiera wszystkie przekazane argumenty, więc porównywanie go do jednego konkretnego nie ma sensu. o2w jednym z atrybutów args, więc robi się tak: `if args.o2w`.